### PR TITLE
[CERT] user notified of successful actions (login, register, reset password). 

### PR DESCRIPTION
**Pull request by MaizeHH**
## Description

User is notified of certain successful actions in the login, register and reset password process. The user gets toast notifications on successful login and email sent successfully in reset password process. The user gets pop-up (dialog) notifications on successful user registered, and password successfully reset. 

## Changes

- ToastNotification component is made to allow easy use of toast notifications (with parameters status, message). 
- DialogNotification component is made to allow easy use of dialog (pop-up) notifications (with parameters status, message). 
- when an email is sent during the password reset process, a toast notification appears. 
- when the user logs into the application, a toast notification appears. 
- when the user successfully resets the password, a dialog notification appears. 
- when the user successfully registers a new user, a dialog notification appears.
- in order to use the Dialog and Toast components, its parent must be wrapped in an AlertNotificationRoot component, which has been added to various files. The tests of these files had to be altered, by mocking this component and the notifications. 

## Related Issues

None. 

## Checklist

- [x] Code has been tested locally and passes all relevant tests.
- [x] Documentation has been updated to reflect the changes, if applicable.
- [x] Code follows the established coding style and guidelines of the project.
- [x] All new and existing tests related to the changes have passed.
- [x] Any necessary dependencies or new packages have been properly documented.
- [x] Pull request title and description are clear and descriptive.
- [x] Reviewers have been assigned to the pull request.
- [x] Any potential security implications have been considered and addressed.
- [x] Performance impact of the changes has been evaluated, if relevant.

## Screenshots (if applicable)

image
image
image
image

## If mobile/frontend pull request, what version of the backend is it stable, and running on? 

Branch: staging

Commit id: 97acc188bc48b67077ed31d97a002fe34c3dc3cd

## Notes for Reviewers

No relevant notes.

### CODE REVIEW

File: eml/components/general/BaseScreen.js
Comment (sofiiagran) [inline eml/components/general/BaseScreen.js:1]: is it a good idea to change in a general component? For example, we use this base screen in our components, can that cause problems?

File: eml/components/general/EducadoModal.js
Comment (sofiiagran) [inline eml/components/general/EducadoModal.js:1]: again, make sure that changing in general components do not cause issues other places where the component is used

File: eml/components/login/RegisterForm.js
Comment (sofiiagran) [inline eml/components/login/RegisterForm.js:1]: I can see some hardcoded colors ...

File: eml/package.json
Comment (sofiiagran) [inline eml/package.json:1]: why you removed this: "react-native-ico-material-design": "^4.1.4",?

**Review by sofiiagran — CHANGES_REQUESTED** [review]
Good work. I have small comments. I have not tested yet, so I will do that later to day.

File: eml/package.json
Comment (MaizeHH) [reply eml/package.json:1]: unsure what prompted this removal but the same package can be found a few lines above so should be fine 👍

**Review by TheMOHD3886 — APPROVED** [review]
Looks aight, good work 👍

File: eml/components/general/EducadoModal.js [hunk 1]
```diff
@@ -22,28 +23,30 @@ export default function EducadoModal(props) {
...
-                  <EducadoLogo fill='fill-black' />
-                </View>
-                <View className="flex flex-row justify-start px-10">
-                  <Text className="text-center text-[24px]">{props.title ? props.title : ""}</Text>
+      <AlertNotificationRoot>
+        <BgLinearGradient>
+          <KeyboardAwareScrollView>
+            <TouchableWithoutFeedback onPress={Keyboard.dismiss}>
+              <View>
+                <View className="flex justify-center pt-[40px]">
```
Comment (ITPini) [inline eml/components/general/EducadoModal.js:31]: Be aware of hardcoded padding top

File: eml/components/general/BaseScreen.js
Comment (MaizeHH) [reply eml/components/general/BaseScreen.js:1]: It caused slight problems when it came to testing, as you would need to mock the AlertNotificationRoot component in every child component (even the ones that don't use any of the notifications). I've changed this now by putting the mocked component in jestSetup.js, such that this mocking does not need to be done repeatedly. I've also done the same for the mocking of KeyboardAwareScrollView.

File: eml/test/screens/register/__snapshots__/register.test.js.snap [hunk 2]
```diff
@@ -566,7 +566,7 @@ exports[`Register screen Login screen renders 1`] = `
                   keyboardType="email-address"
                   onChangeText={[Function]}
                   passwordGuidelines={false}
-                  placeholder="Insira sua e-mail"
+                  placeholder="user@email.com"
```
Comment (JackSMunn) [inline eml/test/screens/register/__snapshots__/register.test.js.snap:569]: This change needs to be discarded.

**Review by JackSMunn — CHANGES_REQUESTED** [review]
GJ

Some snapshot changes that will instantly fail the tests that is currently on dev.

**Review by sofiiagran — CHANGES_REQUESTED** [review]
new merge conflicts since dev have been updated, also I get an error when trying to run this branch (but I do not get the error running dev/stage): 
image

**Comment by MaizeHH** [comment]
> new merge conflicts since dev have been updated, also I get an error when trying to run this branch (but I do not get the error running dev/stage): image

do you use "eas" when building the app? if not, you can delete the "eas" and "projectId" in app.json and then it might work...

**Review by JackSMunn — APPROVED** [review]

**Review by sofiiagran — APPROVED** [review]
I cannot run this and test it, so I can give my approval WITH the demand that someone else test it and makes sure that everything works, including coursescreen/sectionscreen/explore etc.

**Comment by MaizeHH** [comment]
> I cannot run this and test it, so I can give my approval WITH the demand that someone else test it and makes sure that everything works, including coursescreen/sectionscreen/explore etc.

did you try deleting the "eas" and "projectId" from app.json? it might work with those changes

**Pull request merged**